bitcoin/src/blockdata/script.rs

```diff
@@ -400,6 +400,26 @@ impl Script {
       Builder::new()
     }
 
+    /// Returns the public key if this script is P2PK with a valid public key.
+    pub fn p2pk_public_key(&self) -> Option<PublicKey> {
+        PublicKey::from_slice(self.p2pk_pubkey_bytes()?).ok()
+    }
+
+    /// Returns the bytes of the public key if this script is P2PK with a valid public key.
```

Comment:
This doc comment is not correct. It returns the bytes even when the key is invalid (that's the entire point). Maybe worth spelling out though:

> Returns the bytes of the (possibly invalid) public key if this script is P2PK.

Reply:
I like it, adding it in